src/mango.js

```diff
@@ -0,0 +1,232 @@
+  }
+
+  if (config.isV1) return await queryV1(config, indexRef, options)
+  else return await queryV2(config, indexRef, options)
+}
+
+// Internals
+
+const VALUEOPERATORS = ['$eq', '$gt', '$gte', '$lt', '$lte']
+const LOGICOPERATORS = ['$or', '$and', '$not']
+
+/* eslint-disable */
+const MAP_TEMPLATE = (function (doc) {
+  if (doc.docType.toLowerCase() === 'DOCTYPEPLACEHOLDER'){
+    emit(FIELDSPLACEHOLDER, doc)
+  }
+}).toString().replace(/ /g, '').replace(/\n/g, '')
+const COUCHDB_INFINITY = {"\uFFFF": "\uFFFF"}
+const COUCHDB_LOWEST = null
+/* eslint-enable */
+
+// defineIndexV1 is equivalent to defineIndex but only works for V1.
+// It transforms the index fields into a map reduce view.
+async function defineIndexV1 (config, doctype, fields) {
+  let indexName = 'by' + fields.map(capitalize).join('')
+  let indexDefinition = { map: makeMapFunction(doctype, fields), reduce: '_count' }
+  let path = `/request/${doctype}/${indexName}/`
+  await doFetch(config, 'PUT', path, indexDefinition)
+  return { doctype: doctype, type: 'mapreduce', name: indexName, fields: fields }
+}
+
+// defineIndexV1 is equivalent to defineIndex but only works for V1.
+// It transforms the index fields into a map reduce view.
+async function defineIndexV2 (config, doctype, fields) {
+  let path = createPath(config, doctype, '_index')
+  let indexDefinition = {'index': {fields}}
+  let response = await doFetch(config, 'POST', path, indexDefinition)
+  return { doctype: doctype, type: 'mango', name: response.id, fields: fields }
+}
+
+// queryV1 is equivalent to query but only works for V1.
+// It transforms the query into a _views call using makeMapReduceQuery
+async function queryV1 (config, indexRef, options) {
+  if (indexRef.type !== 'mapreduce') {
+    throw new Error('query indexRef should be the return value of defineIndexV1')
+  }
+  if (options.fields) {
+    warn('query fields will be ignored on v1')
+  }
+
+  let path = `/request/${indexRef.doctype}/${indexRef.name}/`
+  let opts = makeMapReduceQuery(indexRef, options)
+  let response = await doFetch(config, 'POST', path, opts)
+  return response.map(r => r.value)
+}
+
+// queryV2 is equivalent to query but only works for V2
+async function queryV2 (config, indexRef, options) {
+  if (indexRef.type !== 'mango') {
+    throw new Error('indexRef should be the return value of defineIndexV2')
+  }
+
+  let opts = {
+    use_index: indexRef.name,
+    fields: options.fields,
+    selector: options.selector,
+    limit: options.limit,
+    since: options.since,
+    sort: indexRef.fields // sort is useless with mango
+  }
+
+  let path = createPath(config, indexRef.doctype, '_find')
+  let response = await doFetch(config, 'POST', path, opts)
+  return response.docs
+}
+
+// misc
+function capitalize (name) {
+  return name.charAt(0).toUpperCase() + name.slice(1)
+}
+
+function makeMapFunction (doctype, fields) {
+  if (Object.prototype.toString.call(fields) !== '[object Array]' ||
```

Comment:
fixed in https://github.com/cozy/cozy-client-js/pull/9/commits/b4712a416c955b7c066da92d0626345e3c1ae1b1 which is part of #9